go/tools/gopackagesdriver/flatpackage.go

```diff
@@ -178,7 +178,11 @@ func (fp *FlatPackage) ResolveImports(resolve ResolvePkgFunc) error {
 	fset := token.NewFileSet()
 
 	for _, file := range fp.CompiledGoFiles {
-		f, err := parser.ParseFile(fset, file, nil, parser.ImportsOnly)
+		var overlayContent []byte
+		if content, ok := overlays[file]; ok {
```

Comment:
```suggestion
		// Only assign overlayContent when an overlay for the file exists, since ParseFile checks by type.
		// If overlay is assigned directly from the map, it will have []byte as type
		// And ParseFile will try to read the (empty) overlay instead of the source file.
		if content, ok := overlays[file]; ok {
```